src/golang/lib/workflow/artifact/preview_cache.go

```diff
@@ -0,0 +1,91 @@
+package artifact
+
+import (
+	"context"
+
+	"github.com/aqueducthq/aqueduct/lib/collections/shared"
+	"github.com/aqueducthq/aqueduct/lib/database"
+	"github.com/aqueducthq/aqueduct/lib/workflow/utils"
+	"github.com/google/uuid"
+	lru "github.com/hashicorp/golang-lru"
+)
+
+const maxNumEntries = 200
+
+// PreviewCacheEntry is the object that a user of this cache will see when fetching.
+type PreviewCacheEntry struct {
+	ArtifactContentPath  string
+	ArtifactMetadataPath string
+	OpMetadataPath       string
+}
+
+type PreviewCacheManager interface {
+	// Attempts to fetch the cache entries for the given artifacts, returning them
+	// in a map for easy lookup. An entry that does not exist will not be present in the
+	// returned map. Also returns a boolean indicating whether all artifacts were found in the
+	// cache.
+	GetMulti(ctx context.Context, logicalIDs []uuid.UUID) (bool, map[uuid.UUID]PreviewCacheEntry, error)
+
+	// Writes the given entries into the cache. If entries already exist with the same artifact ID,
+	// they will be deleted before the write takes place.
```

Comment:
I didn't find the logic where we are GCing the previous paths' content.

Reply:
We don't GC, but instead set a limit on the cache size and let the next server start clean things up: https://github.com/aqueducthq/aqueduct/pull/338. Do you think we should also add a background thread to evicts entries after some time?

Reply:
Discussed: fixed.